### ISSUE TITLE
Python build is failing with Pipfile.lock and null markers

### ISSUE THREAD

**Issue by rclement** [zeit/now#8933]
## Issue

When try to build a Python serverless project on Vercel, the build is sometimes failing with the following error:

```
[20:15:35.199] Retrieving list of deployment files...
[20:15:35.473] Previous build caches not available
[20:15:37.624] Downloading 13 deployment files...
[20:15:38.535] Running "vercel build"
[20:15:39.191] Vercel CLI 28.5.3
[20:15:39.361] WARN! Due to `builds` existing in your configuration file, the Build and Development Settings defined in your Project Settings will not apply. Learn More: https://vercel.link/unused-build-settings
[20:15:39.460] Installing required dependencies...
[20:15:41.462] Failed to run "pipfile2req"
[20:15:41.522] Error: Command failed: /tmp/1d640a54/bin/pipfile2req
[20:15:41.523] Traceback (most recent call last):
[20:15:41.523]   File "/tmp/1d640a54/bin/pipfile2req", line 8, in <module>
[20:15:41.523]     sys.exit(main())
[20:15:41.523]   File "/tmp/1d640a54/pipfile2req/__init__.py", line 112, in main
[20:15:41.523]     lines = convert_pipfile_or_lock(
[20:15:41.523]   File "/tmp/1d640a54/pipfile2req/__init__.py", line 96, in convert_pipfile_or_lock
[20:15:41.524]     lines = [
[20:15:41.524]   File "/tmp/1d640a54/pipfile2req/__init__.py", line 97, in <listcomp>
[20:15:41.524]     requirement_from_pipfile(name, package, hashes)
[20:15:41.524]   File "/tmp/1d640a54/pipfile2req/requirements.py", line 10, in requirement_from_pipfile
[20:15:41.524]     return Requirement.parse(name, package).as_line(include_hashes)
[20:15:41.524]   File "/tmp/1d640a54/pipfile2req/requirements.py", line 97, in parse
[20:15:41.524]     kwargs["markers"] = _merge_markers(markers)
[20:15:41.525]   File "/tmp/1d640a54/pipfile2req/requirements.py", line 14, in _merge_markers
[20:15:41.525]     result = Marker(markers[0])
[20:15:41.525]   File "/tmp/1d640a54/packaging/markers.py", line 278, in __init__
[20:15:41.525]     self._markers = _coerce_parse_result(MARKER.parseString(marker))
[20:15:41.525]   File "/tmp/1d640a54/pyparsing/core.py", line 1129, in parse_string
[20:15:41.525]     instring = instring.expandtabs()
[20:15:41.525] AttributeError: 'NoneType' object has no attribute 'expandtabs'
[20:15:41.525] 
[20:15:41.526] 
```

## Investigations

Investigation on my part across multiple project with the same error showed the following:

- `Pipfile.lock` (generated by `pipenv`) is containing `"markers": null` for some packages
- `pipfile2req` (used by Vercel Python runtime) is trying to parse markers without checking for null values thus failing

## Potential solutions

- Submit a PR to [`pipfile2req`](https://github.com/frostming/pipfile-requirements) to fix the behavior
- Instead of `pipfile2req`, use `pipenv requirements` within Vercel Python runtime to generate the `requirements.txt` file

**Comment by andymckay** [comment]
Thank you @rclement. Do you have an example requirements.txt that produces `"markers": null` so we can reproduce this?

**Comment by rclement** [comment]
@andymckay you can look-up currently failing PRs on the following repos:

- https://github.com/rclement/mailer/pulls
- https://github.com/rclement/business-card-generator/pulls

Those failing PRs are all coming from RenovateBot generating `"markers": null` in the `Pipfile.lock file`.

**Comment by andymckay** [comment]
I've been able to reproduce this locally too. I'm suspecting that the problem is that `pipenv` should not be generating `marker=null` in the first place. For example in your case the line:

```
[packages]
pydantic = {extras = ["email"],version = "==1.10.2"}
```

Generates null for the marker.

```
[packages]
pydantic = "==1.10.2"
```

Works for me just fine. I admit at this point I'm not sure what `extras = ["email"]` does, yet 😄

**Comment by andymckay** [comment]
Ah so looking at https://pydantic-docs.helpmanual.io/install/#optional-dependencies, I can see that this pulls in `email-validator` so making my pipfile:

```
pydantic = "==1.10.2"
email-validator = "==1.3.0"
```

Works great for me and I recommend we bounce this issue up to `pipenv`.

**Comment by andymckay** [comment]
pipenv issue: https://github.com/vercel/vercel/issues/8933

**Comment by rclement** [comment]
@andymckay thanks for your findings! Indeed this seems to be an issue with pipenv not handling markers properly. However, generating `requirements.txt` using the native `pipenv requirements` command works great whereas using `pipfile2req` does not handle every edge-cases.

I just take the liberty of linking the right `pipenv` issue in here:
https://github.com/pypa/pipenv/issues/5486

**Comment by andymckay** [comment]
Oops, thanks for spotting that.

**Comment by rclement** [comment]
Issue is fixed starting with `pipenv` version `2022.12.17`: https://github.com/pypa/pipenv/releases/tag/v2022.12.17

FYI: Renovatebot seems to have picked up the latest version of `pipenv` as null markers are not appearing in `Pipfile.lock` anymore
